pkgs/sops-install-secrets/main.go

```diff
@@ -1196,21 +1196,59 @@ func replaceRuntimeDir(path, rundir string) (ret string) {
 	return
 }
 
-func writeTemplates(targetDir string, templates []template, keysGID int, userMode bool) error {
-	for _, template := range templates {
-		fp := filepath.Join(targetDir, template.Name)
+func writeTemplate(targetDir string, template template, keysGID int, userMode bool) error {
+	templatePath := filepath.Join(targetDir, template.Name)
 
-		if err := createParentDirs(targetDir, template.Name, keysGID, userMode); err != nil {
-			return err
+	if err := createParentDirs(targetDir, template.Name, keysGID, userMode); err != nil {
+		return err
+	}
+
+	dir := filepath.Dir(templatePath)
+	tempFile, err := os.CreateTemp(dir, "sops-*")
+	tempfileRemoved := false
+	if err != nil {
+		return fmt.Errorf("cannot create temporary file in directory %s: %w", dir, err)
+	}
+	defer func() {
+		tempFile.Close() // noop if already closed
+
+		if !tempfileRemoved {
+			if err := os.Remove(tempFile.Name()); err != nil {
+				fmt.Fprintf(os.Stderr, "failed to remove temporary file %s: %s\n", tempFile.Name(),
+					err)
+			}
 		}
+	}()
```

Comment:
I like to reduce nested ifs and keep the control flow as linear as possible. 

```suggestion
        if tempFileRemoved {
            // nothing to do
            return
        }
		
        if err := os.Remove(tempFile.Name()); err != nil {
        	fmt.Fprintf(os.Stderr, "failed to remove temporary file %s: %s\n", tempFile.Name(),
        		err)
        }		
	}()
```